cocotb/share/lib/vhpi/VhpiCbHdl.cpp

```diff
@@ -268,7 +269,7 @@ int VhpiSignalObjHdl::initialise(std::string &name, std::string &fq_name) {
         case vhpiStrVal: {
             m_indexable = true;
             m_num_elems = vhpi_get(vhpiSizeP, handle);
-            m_value.bufSize = (m_num_elems)*sizeof(vhpiCharT) + 1;
+            m_value.bufSize = (m_num_elems)*static_cast<int>(sizeof(vhpiCharT)) + 1;
```

Comment:
This cast is correct only on `IUS` - on other simulators, `bufSize` is `size_t`. Perhaps use the somewhat clumsy:
```suggestion
            m_value.bufSize = static_cast<decltype(m_value.bufSize)>(
                m_num_elems * sizeof(vhpiCharT) + 1
            );
```

Reply:
I would prefer to use that everywhere in my changes. Are we definitely not averse to usage of `decltype` and `auto`?

Reply:
Well, we need #1149 to go in first, to add `std=c++11` (or another reduced patch which does so).

Reply:
#1137 Already did that. We just don't have coding standards for newer C++ functionality.

Reply:
There are very few ways to write correct code here:
* Use `decltype`
* Use `#ifdef`, and repeat the logic in the `vhpi_user.h` header

I don't think standards are necessary here. Let's use `decltype` because it's the only tool appropriate for the job (and add a comment saying that the type of `bufSize` varies by simulator).